src/ipc/ipc4/handler.c

```diff
@@ -425,6 +427,21 @@ static int ipc4_set_pipeline_state(struct ipc4_message_request *ipc4)
 		ppl_id = &id;
 	}
 
+	for (i = 0; i < ppl_count; i++) {
+		ppl_icd = ipc_get_comp_by_ppl_id(ipc, COMP_TYPE_PIPELINE, ppl_id[i]);
+		if (!ppl_icd) {
+			tr_err(&ipc_tr, "ipc: comp %d not found", ppl_id[i]);
+			return IPC4_INVALID_RESOURCE_ID;
+		}
+
+		if (i) {
+			if (ppl_icd->core != idx)
+				use_idc = true;
+		} else {
+			idx = ppl_icd->core;
+		}
```

Comment:
if else parentheses should not need

Reply:
I think this is good here for readability.

Reply:
it is not only for readability, without them the `else` would become ambiguous, IIRC the compiler issues a warning in such cases, and in fact it would be misinterpreted as referring to the innermost `if`, i.e. removing braces here would actually mean
```
	if (i)
		if (ppl_icd->core != idx) {
			use_idc = true;
		} else {
			idx = ppl_icd->core;
		}
```
which would be very different to what is intended.